src/V3EmitCFunc.h

```diff
@@ -666,15 +666,67 @@ class EmitCFunc VL_NOT_FINAL : public EmitCConstInit {
         putsQuoted(nodep->linescov());
         puts(");\n");
     }
+    void visit(AstCoverToggleDecl* nodep) override {
+        putns(nodep, "vlSelf->__vlCoverToggleInsert(");  // As Declared in emitCoverageDecl
+        puts(cvtToStr(nodep->range().right()));
+        puts(", ");
+        puts(cvtToStr(nodep->range().left()));
+        puts(", ");
+        puts(cvtToStr(nodep->range().ranged()));
+        puts(", ");
+        puts("&(vlSymsp->__Vcoverage[");
+        puts(cvtToStr(nodep->dataDeclThisp()->binNum()));
+        puts("])");
+        // If this isn't the first instantiation of this module under this
+        // design, don't really count the bucket, and rely on verilator_cov to
+        // aggregate counts.  This is because Verilator combines all
+        // hierarchies itself, and if verilator_cov also did it, you'd end up
+        // with (number-of-instant) times too many counts in this bin.
```

Comment:
I don't understand this.  Each hier should be counted separately, why is the first one different?